confgenerator/testdata/valid/linux/metrics-default_overrides_disable_all/golden_otel.conf

```diff
@@ -32,7 +32,7 @@ processors:
       exclude:
         match_type: regexp
         metric_names:
-        - ^.*$
+        - "^agent\\.googleapis\\.com/.*$$"
```

Comment:
I don't think that it needs "$$"

Reply:
I believe the `$$` is needed to properly handle the future case where it doesn't end in `/*`.

Reply:
In regexp mode, the metrics names are just any regular expressions.  Why it will stop working when it not ends with /.*?

Reply:
I've tested with the single `$` — it looks like only metricstransform needed the escaping.

Reply:
There's nothing specific to `metricstransform` - https://opentelemetry.io/docs/collector/configuration/#configuration-environment-variables

Reply:
Ah. That explains why it worked without escaping in this specific case — `os.Expand` ignores trailing `$` signs. Reverted to the escaping commit.